### PR TITLE
Fix/handled error stack trace

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code? <- (Can add unit tests at a later time; working E2E and need to include in release)
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
Fixes a behavior where stack traces from handled errors are missing frames.

### CODE REVIEW

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.WindowsDesktop/Utils/StackTraceHelper.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        public static string GenerateFullStackTrace(Exception e)
+        {
+            if (string.IsNullOrEmpty(e.StackTrace))
+            {
+                return e.StackTrace;
+            }
+            var exceptionStackTrace = new System.Diagnostics.StackTrace(e, true);
+
+            // Generate current stack trace. Skip three frames to avoid showing SDK code.
+            var currentStackTrace = new System.Diagnostics.StackTrace(3, true);
```
Comment (jwallra) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.WindowsDesktop/Utils/StackTraceHelper.cs]: Can we make a note of what the three frames skipped are?  Right now it looks like a magic number

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.WindowsDesktop/Utils/StackTraceHelper.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+using System;
+using System.Linq;
+
+namespace Microsoft.AppCenter.Crashes.Utils
+{
+    public class StackTraceHelper
+    {
+        // Exceptions don't always have complete stack traces, so they must be augmented.
+        public static string GenerateFullStackTrace(Exception e)
```
Comment (jwallra) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.WindowsDesktop/Utils/StackTraceHelper.cs]: I'm a bit confused as to why this is duplicated here from the one in Crashes.cs.  Does it have windows desktop-specific code?  If so, what code is not shared?

File: CHANGELOG.md [hunk 3]
```diff
@@ -37,9 +37,10 @@ This preview version removes Crashes support for UWP. The App Center backend is
 #### iOS
 
 * **[Fix]** Fix possible deadlock in `Crashes.TrackError`.
+* **[Fix]** Fix an issue where stack traces for exceptions passed to `Crashes.TrackError` were incomplete.
 
 #### Android
-
+* **[Fix]** Fix an issue where stack traces for exceptions passed to `Crashes.TrackError` were incomplete.
```
Comment (guperrot) [inline CHANGELOG.md]: Move statement outside of platform, all are impacted.

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Android/Crashes.cs [hunk 4]
```diff
@@ -151,35 +152,85 @@ static ModelException GenerateModelException(Exception exception, bool structure
...
                 {
-                    GenerateModelException(exception.InnerException, structuredFrames)
+                    GenerateModelException(exception.InnerException)
                 };
             }
             return modelException;
         }
 
-        static IList<ModelStackFrame> GenerateModelStackFrames(StackTrace stackTrace)
+        // Exceptions don't always have complete stack traces, so they must be augmented.
```
Comment (guperrot) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Android/Crashes.cs]: Please comment that crashes don't need stack trace augmenting (all platforms).

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs [hunk 5]
```diff
@@ -188,6 +188,11 @@ void AggregateException(object sender, EventArgs e)
             HandleOrThrow(() => throw PrepareException());
         }
 
+        void AmbiguousException(object sender, EventArgs e)
+        {
+            BeginOrEndAmbigousCase();
```
Comment (clpolet) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs]: nit typo:
```suggestion
            BeginOrEndAmbiguousCase();
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -26,6 +26,8 @@ This preview version removes Crashes support for UWP. The App Center backend is
 
 ### App Center Crashes
 
+* **[Fix]** Fix an issue where stack traces for exceptions passed to `Crashes.TrackError` were incomplete.
```
Comment (Jamminroot) [inline CHANGELOG.md]: This line might need to be updated now (moved to different version's section)

Comment (MatkovIvan) [reply CHANGELOG.md]: Yeah, please move it to the current version

**Comment by michaelsmitha** [comment]
I believe the current state of this is that in testing we found that, while the stack traces are much better in some cases - they are much worse in others, so more work is required.

**Comment by guperrot** [comment]
symbolication improvements will be done server side.
there will likely be a sdk change to send the additional frames though.

**Pull request closed without merging**